mne_sandbox/connectivity/cfc.py

```diff
@@ -0,0 +1,564 @@
+        time-locked to peaks of the low-frequency phase.
+    data_phase : np.array
+        The mean low-frequency signal, phase-locked to low-frequency phase
+        peaks.
+    times : np.array
+        The times before / after each phase peak.
+    """
+    sfreq = inst.info['sfreq']
+    # Pull the amplitudes/phases using Morlet
+    data_ph, data_amp = _pull_data(inst, ix_ph, ix_amp)
+    angle_ph, band_ph, amp = _extract_phase_and_amp(
+        data_ph, data_amp, sfreq, freqs_phase, freqs_amp)
+
+    angle_ph = angle_ph.mean(0)  # Mean across freq bands
+    band_ph = band_ph.mean(0)
+
+    # Find peaks in the phase for time-locking
+    phase_peaks, vals = peak_finder.peak_finder(angle_ph)
+    ixmin, ixmax = [t * sfreq for t in [tmin, tmax]]
+    # Remove peaks w/o buffer
+    phase_peaks = phase_peaks[(phase_peaks > np.abs(ixmin)) *
+                              (phase_peaks < len(angle_ph) - ixmax)]
+
+    if mask_times is not None:
+        # Set datapoints outside out times to nan so we can drop later
+        if len(mask_times) != angle_ph.shape[-1]:
+            raise ValueError('mask_times must be == in length to data')
+        band_ph[..., ~mask_times] = np.nan
+
+    data_phase, times, msk_window = _raw_to_epochs_array(
+        band_ph[np.newaxis, :], sfreq, phase_peaks, tmin, tmax)
+    data_amp, times, msk_window = _raw_to_epochs_array(
+        amp, sfreq, phase_peaks, tmin, tmax)
+    data_phase = data_phase.squeeze()
+    data_amp = data_amp.squeeze()
+
+    # Drop any peak events where there was a nan
+    keep_rows = np.where(~np.isnan(data_phase).any(-1))[0]
+    data_phase = data_phase[keep_rows, ...]
+    data_amp = data_amp[keep_rows, ...]
+
+    # Average across phase peak events
+    data_amp = data_amp.mean(0)
+    data_phase = data_phase.mean(0)
+    return data_amp, data_phase, times
+
+
+def phase_binned_amplitude(inst, freqs_phase, freqs_amp,
+                           ix_ph, ix_amp, n_bins=20, mask_times=None):
+    """Calculate amplitude of one signal in sub-ranges of phase for another.
+
+    Parameters
+    ----------
+    inst : instance of mne.Epochs or mne.io.Raw
+        The data to be used in phase locking computation
+    freqs_phase : np.array
+        The frequencies to use in phase calculation. The phase of each
+        frequency will be averaged together.
+    freqs_amp : np.array
+        The frequencies to use in amplitude calculation. The amplitude
+        of each frequency will be averaged together.
+    ix_ph : int
+        The index of the signal to be used for phase calculation
+    ix_amp : int
+        The index of the signal to be used for amplitude calculation
+    n_bins : int
+        The number of bins to use when grouping amplitudes. Each bin will
+        have size (2 * np.pi) / n_bins.
+    mask_times : np.array, dtype bool, shape (inst.n_times,)
+        If inst is an instance of Raw, this will only include times contained
```

Comment:
ahh I see, I was confused because when I google 'backticks' it sent me to
the page for 'backslash'. Thanks for clarifying

On Fri, May 13, 2016 at 4:25 PM, Jean-Rémi KING <notifications@github.com>
wrote:

> In mne_sandbox/connectivity/cfc.py
> <https://github.com/mne-tools/mne-sandbox/pull/7#discussion_r63263841>:
>
> > +        The data to be used in phase locking computation
> > +    freqs_phase : np.array
> > +        The frequencies to use in phase calculation. The phase of each
> > +        frequency will be averaged together.
> > +    freqs_amp : np.array
> > +        The frequencies to use in amplitude calculation. The amplitude
> > +        of each frequency will be averaged together.
> > +    ix_ph : int
> > +        The index of the signal to be used for phase calculation
> > +    ix_amp : int
> > +        The index of the signal to be used for amplitude calculation
> > +    n_bins : int
> > +        The number of bins to use when grouping amplitudes. Each bin will
> > +        have size (2 * np.pi) / n_bins.
> > +    mask_times : np.array, dtype bool, shape (inst.n_times,)
> > +        If inst is an instance of Raw, this will only include times contained
>
> inst is a variable, so you need to surround it with backticks: e.g.
>
> "If `inst` is an instance of Raw, this will only"
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-sandbox/pull/7/files/88eb3c34c76f70ef766893583e5cfd66124849df#r63263841>
>